x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/gen/pipeline/MultiPipe.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.gen.pipeline;
+
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.gen.processor.Processor;
+import org.elasticsearch.xpack.sql.tree.Location;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class MultiPipe extends Pipe {
+
+    protected MultiPipe(Location location, Expression expression, List<Pipe> children) {
+        super(location, expression, children);
+    }
+
+    @Override
+    public Processor asProcessor() {
+        List<Processor> procs = new ArrayList<>();
+        for (Pipe pipe : children()) {
+            procs.add(pipe.asProcessor());
+        }
+
+        return asProcessor(procs);
+    }
+
+    public abstract Processor asProcessor(List<Processor> procs);
+}
```

Comment:
No, because this check already exists. From `Node#equals`:
```
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
```

Reply:
ah ok, overlooked that!